### PR TITLE
Add support for deserializing sets in gremlin-javascript

### PR DESCRIPTION
**Pull request by kenhuuu**
This is marked as draft as it is a breaking change since gremlin-javascript used to deserialize sets as arrays. This feature would need to go into either 3.8 or 4.0.

This also builds on top of https://github.com/apache/tinkerpop/pull/2279 right now but will be retargeted to master after 3.7 has been split away.

VOTE +1

### CODE REVIEW

**Review by cubeddu — COMMENTED** [review]
JS changes LGTM @kenhuuu

File: gremlin-javascript/src/main/javascript/gremlin-javascript/lib/structure/io/binary/internals/SetSerializer.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+/**
+ * @author Igor Ostapenko
```
Comment (vkagamlyk) [inline gremlin-javascript/src/main/javascript/gremlin-javascript/lib/structure/io/binary/internals/SetSerializer.js:21]: no.

Comment (kenhuuu) [reply gremlin-javascript/src/main/javascript/gremlin-javascript/lib/structure/io/binary/internals/SetSerializer.js:21]: I wanted to give them credit for this as the major of this file is copied from ArraySerializer.js

**Comment by Cole-Greer** [comment]
LGTM VOTE +1

**Pull request merged**